apps/framework-docs/src/pages/moose/olap/db-pull.mdx

```diff
@@ -0,0 +1,143 @@
+
+- **External tables changed remotely**: a DBA, CDC, or ETL pipeline updated schema.
+- **Keep types in sync**: update generated models without touching fully-managed tables.
+- **Safe by design**: does not modify the database or your managed models.
+
+<Callout type="info">This is a read-only sync for your code models. For concepts and modeling guidance, see [External Tables](/moose/olap/external-tables). To bootstrap a project from an existing DB, see [Initialize from ClickHouse](/moose/getting-started/from-clickhouse).</Callout>
+
+## Requirements
+
+- Tables are defined with <TypeScript inline>`lifeCycle: EXTERNALLY_MANAGED`</TypeScript><Python inline>`life_cycle=LifeCycle.EXTERNALLY_MANAGED`</Python>
+- A ClickHouse connection string (native or HTTP/S)
+
+## Connection strings
+
+`db pull` accepts both native and HTTP(S) URLs. Native strings are automatically converted to HTTP(S) with the appropriate ports.
+
+Examples:
+
+```bash filename="Terminal" copy
+# Native (auto-converted to HTTPS + 8443)
+moose db pull --connection-string "clickhouse://explorer@play.clickhouse.com:9440/default"
+
+# HTTPS (explicit database via query param)
+moose db pull --connection-string "https://play.clickhouse.com/?user=explorer&database=default"
+
+# Local HTTP
+moose db pull --connection-string "http://localhost:8123/?user=default&database=default"
+```
+
+## What gets written
+
+- TypeScript: `app/externalModels.ts`
+- Python: `app/external_models.py`
+
+The file contains only the tables you marked as `EXTERNALLY_MANAGED` and starts with an auto-generated header.
+
+Important:
+
+- The generator always writes to the specified output path, overwriting the file each time. If you pass `--file-path`, that path is regenerated on every run.
+- We recommend keeping external models at the default paths above. Moose’s init/remote import flows add imports to <TypeScript inline>`app/index.ts`</TypeScript><Python inline>`app/main.py`</Python> that expect these defaults. If you change the output path, make sure your main file imports the new location so the models are loaded.
+
+## How it works
+
+When you run `db pull` the CLI does the following:
+
+- Loads your project’s infrastructure map and identifies tables marked as `EXTERNALLY_MANAGED`.
+- Connects to the remote ClickHouse specified by `--connection-string` and introspects the live schemas for those tables.
+- Regenerates a single external models file that mirrors the remote schema.
+- Does not change any fully managed tables, your <TypeScript inline>`app/index.ts`</TypeScript><Python inline>`app/main.py`</Python>, or the database itself.
+- Creates a small git commit if the generated file changed, so you can review and share the update.
+
+### TypeScript output example
+
+<TypeScript>
+```ts filename="app/externalModels.ts"
+// AUTO-GENERATED FILE. DO NOT EDIT.
+// This file will be replaced when you run `moose db pull`.
+
+// ...typed table definitions matching remote EXTERNALLY_MANAGED tables...
```

Comment:
not sure why its useful to give an "example" here that doesn't actually have any real example code in it?